lib/render/mayaToHydra/plugin.cpp

```diff
@@ -0,0 +1,123 @@
+//
+// Copyright 2019 Luma Pictures
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include <maya/MFnPlugin.h>
+
+#include "renderGlobals.h"
+#include "renderOverride.h"
+#include "usdPreviewSurface.h"
+#include "viewCommand.h"
+
+#include <stdlib.h>
+
+#include <pxr/base/plug/plugin.h>
+#include <pxr/base/plug/registry.h>
+#include <pxr/base/tf/envSetting.h>
+
+#include "../../usd/hdMaya/adapters/adapter.h"
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+TF_DEFINE_ENV_SETTING(
+    MTOH_ENABLE_USD_PREVIEW_SURFACE_NODE, true,
+    "Enables the registration of the UsdPreviewSurface node."
+    "This is not required with newer version of usdMaya.");
+
+namespace {
+bool _enableUsdPreviewSurface = true;
+}
+
+PLUGIN_EXPORT MStatus initializePlugin(MObject obj) {
+    MStatus ret = MS::kSuccess;
+
+    ret = HdMayaAdapter::Initialize();
+    if (!ret) { return ret; }
+
+    // For now this is required for the HdSt backed to use lights.
+    // putenv requires char* and I'm not willing to use const cast!
+    constexpr const char* envVarSet = "USDIMAGING_ENABLE_SCENE_LIGHTS=1";
+    std::vector<char> envVarData;
+    envVarData.resize(strlen(envVarSet) + 1);
+    sprintf(envVarData.data(), "%s", envVarSet);
```

Comment:
Addressed in:

https://github.com/Autodesk/maya-usd/pull/231/commits/4d3619fccdb679c0bd0deccc2476a3f2964069ab

Note that I used `snprintf`, not `sprintf_s`, since the latter isn't available on Linux.

Also, I fixed a pair of other usages (though they weren't in hdmaya).